numpydoc/tests/test_full.py

```diff
@@ -65,3 +66,23 @@ def test_my_function(sphinx_app):
     assert '*args' in html
     # check xref (iterable should link using xref):
     assert 'glossary.html#term-iterable' in html
+
+
+def test_reference(sphinx_app):
+    """Test for bad references e"""
+    out_dir = sphinx_app.outdir
+    html_files = [
+        ["index.html"],
+        ["generated", "numpydoc_test_module.my_function.html"],
+        ["generated", "numpydoc_test_module.MyClass.html"],
+    ]
+
+    for html_file in html_files:
+        html_file = op.join(out_dir, *html_file)
+
+        with open(html_file, 'r') as fid:
+            html = fid.read()
+
+        reference_list = re.findall(r'<a class="fn-backref" href="\#id\d+">(.*)<\/a>', html)
```

Comment:
```suggestion
        reference_list = re.findall(r'<a class="fn-backref" href="\#id\d+">(.*)<\/a>', html)
        assert len(reference_list) == 3
```
Otherwise this could be an empty list and thus not testing what it means to test

Reply:
I like it. It is only index.html which have 3 items in the `reference_list`. I have added the check to the test.